src/models/metrics.py

```diff
@@ -0,0 +1,70 @@
+import numpy as np
+import pandas as pd
+import matplotlib.pyplot as plt
+
+
+# Approximated differentiable SMAPE
+def differentiable_smape(true, predicted):
+    epsilon = 0.1
+    true_o = true
+    pred_o = predicted
+    summ = np.maximum(np.abs(true_o) + np.abs(pred_o) + epsilon, 0.5 + epsilon)
+    smape = np.abs(pred_o - true_o) / summ
+    return smape
+
+
+# SMAPE, rounded up to the closest integer
+def rounded_smape(true, predicted):
+    true_o = np.int(np.round(true))
+    pred_o = np.round(predicted).astype(np.int32)
+    summ = np.abs(true_o) + np.abs(pred_o)
+    smape = np.where(summ==0, 0, np.abs(pred_o - true_o) / summ)
+    return smape
+
+
+# SMAPE stardard definition
+def smape(true, predicted):
+    true_o = true
+    pred_o = predicted
+    summ = np.abs(true_o) + np.abs(pred_o)
+    smape = np.where(summ == 0, 0, np.abs(pred_o - true_o) / summ)
+    return smape
+
+
+# MAE standard definition
+def mape(true, predicted):
+    true_o = true
+    pred_o = predicted
+    denom = np.abs(true_o)
+    mape = np.where(denom == 0, 0, np.abs(pred_o - true_o) / denom)
+    return mape
+
+
+# MAE on log1p
+def mape1p(true, predicted):
+    epsilon = 0.1
+    true_o = np.log1p(true + epsilon)
+    pred_o = np.log1p(predicted + epsilon)
+    error = np.abs(true_o - pred_o)/np.abs(true_o)
+    return error
+
+
+# input pd.series in which index = time stamps, value = time series values
+# actual_series and pred_series should have exactly matched timestamps
+def plot_metrics(actual_series, pred_series):
+    import pandas as pd
+    import numpy as np
+    import matplotlib.pyplot as plt
```

Comment:
its generally bad practice to have import statements inside functions unless you really need to do some fancy dependency injection or there is some sort of speed concern where you want to lazily load the libraries at this point.

The biggest reason is now - everytime we call this function, we are going to reload the libraries.
Its fine (and preferred) just to have them at the top of the file.